src/libasr/pass/intrinsic_function.cpp

```diff
@@ -71,9 +72,19 @@ class ReplaceIntrinsicFunctions: public ASR::BaseExprReplacer<ReplaceIntrinsicFu
         }
         ASR::ttype_t* type = nullptr;
         type = ASRUtils::extract_type(x->m_type);
-        ASR::expr_t* current_expr_ = instantiate_function(al, x->base.base.loc,
+        ASR::expr_t* current_expr_ = nullptr;
+         try {
+            current_expr_ = instantiate_function(al, x->base.base.loc,
             global_scope, arg_types, type, new_args, x->m_overload_id);
-        *current_expr = current_expr_;
+            *current_expr = current_expr_;
+        } catch (const LCompilersException& e) {
+            global_diagnostics->add(diag::Diagnostic(
+                std::string(e.what()),
+                diag::Level::Error, diag::Stage::Semantic, {
```

Comment:
```suggestion
                diag::Level::Error, diag::Stage::XX, {
```

Do you think `Semantic` is good here? or we need to have some other name?

Reply:
We can change these if required.